Sources/DataStructures/Graph/Protocols/GraphProtocol.swift

```diff
@@ -62,6 +62,25 @@ extension GraphProtocol {
     public func neighbors(of source: Node, in nodes: Set<Node>? = nil) -> Set<Node> {
         return (nodes ?? self.nodes).filter { edges.contains(Edge(source,$0)) }
     }
+    
+    /// - Returns: A set of edges that contain the given `node` (either incident or
+    /// outgoing).
+    @inlinable
+    public func edges(containing node: Node) -> Set<Edge> {
+        return edges(from: node).union(edges(to: node))
+    }
+    
+    /// - Returns: A set of edges outgoing from the given `source`.
+    @inlinable
+    public func edges(from source: Node) -> Set<Edge> {
+        return Set(neighbors(of: source).map { Edge(source, $0) })
```

Comment:
This is great. I wonder if there would be any benefit to using `neighbors(of: source).lazy.map { ... }`

Reply:
Ah, cool!